mmcv/runner/hooks/optimizer.py

```diff
@@ -41,6 +62,32 @@ def after_train_iter(self, runner):
                                          runner.outputs['num_samples'])
         runner.optimizer.step()
 
+    def detect_anomalous_parameters(self, loss, runner):
+        logger = runner.logger
+        parameters_in_graph = set()
+        visited = set()
+
+        def traverse(grad_fn):
+            if grad_fn is None:
+                return
+            if grad_fn not in visited:
+                visited.add(grad_fn)
+            else:
+                return
+            if hasattr(grad_fn, 'variable'):
+                parameters_in_graph.add(grad_fn.variable)
+            parents = grad_fn.next_functions
+            if parents is not None:
+                for parent in parents:
+                    grad_fn = parent[0]
+                    traverse(grad_fn)
```

Comment:
insert these lines in `if grad_fn not in visited` statement after `visited.add(grad_fn)`